_source/_posts/2023-07-28-scim-workshop.md

```diff
@@ -0,0 +1,1144 @@
+        data: {
+            name: 'Somnus',
+            email: 'somnus@portal.example',
+            password: 'correct horse battery staple',
+            orgId: org.id,
+            externalId: '22',
+            active: true
+        },
+    })
+    console.log('Created user Somnus', somnusUser)
+
+    const trinityUser = await prisma.user.create({
+        data: {
+            name: 'Trinity',
+            email: 'trinity@portal.example',
+            password: 'Zion',
+            orgId: org.id,
+            externalId: '23',
+            active: true
+        },
+    })
+    console.log('Created user Trinity', trinityUser)
+}
+
+main()
+    .then(async () => {
+        await prisma.$disconnect()
+    })
+    .catch(async (e) => {
+        console.error(e)
+        await prisma.$disconnect()
+        process.exit(1)
+    })
+```
+Now we are ready to seed the database. If you already added records to the database, you can run `npx prisma migrate reset` before running `npm run init-db`.
+
+A neat feature with Prisma is the option to view the user table locally. To do this, go to the root of this workshop folder, and run `npx prisma studio`. You should be redirected to a webpage where you can see all the tables you've created. 
+
+
+## Add a SCIM file and create a scimRoute
+
+For maintainability, we should keep the code we add for SCIM support in one place. Let's create a SCIM file, `scim.ts`, in`apps/src/assets`, and import it in `main.ts`. As a result, your `scim.ts` file will look like this: 
+
+```
+import { Router } from 'express';
+export const scimRoute = Router();
+```
+
+And you'll import the routes in `main.ts` like this:
+```
+import { scimRoute } from './scim';
+app.use('/scim/v2', scimRoute);
+```
+
+Note: `/scim/v2` will append to every SCIM route. You can change this static path as needed, as there isn't a specific URL path explicitly mentioned by the SCIM spec. However, [Okta's SCIM Docs](https://developer.okta.com/docs/guides/scim-provisioning-integration-prepare/main/#base-url) recommend using `/scim/v2` unless unusual factors in your environment necessitate a different URL. 
+
+## Build the SCIM interface:
+
+We'll need to build each CRUD endpoint required by the SCIM spec and format the responses in JSON. After that, we'll test our functions with Postman to see that they provide the responses required by the spec. 
+
+We'll need to use express and Prisma in `scim.ts`, so import them:
+
+```
+import express from 'express';
+import { Prisma, PrismaClient } from '@prisma/client';
+```
+
+Some SCIM endpoints in `scim.ts` will return user information from the Todo app's database. To easily retrieve this information from the SCIM endpoints, implement an `IUserSchema` interface and instantiate it as the defaultUserSchema. The information in the `IUserSchema` matches the SCIM spec, as you'll see soon.  
+
+```
+const prisma = new PrismaClient();
+
+// To funnel users into their designated orgs
+const ORG_ID = 1;
```

Comment:
Why is this hardcoded to 1? Needs an explanation of what's going on here. Is this a placeholder value that you should change in production?